route_info/src/lib.rs

```diff
@@ -64,11 +67,27 @@ impl RouteInfo {
             .closest_intersection
             .closest_pt(pt2, threshold)
             .ok_or_else(|| err_to_js("no intersection close to last pt"))?;
-        let i1_name = self.map.get_i(i1).name(lang, &self.map);
-        let i2_name = self.map.get_i(i2).name(lang, &self.map);
 
+        let i1_name = self.name_intersection(i1);
+        let i2_name = self.name_intersection(i2);
         Ok(format!("Route from {i1_name} to {i2_name}"))
     }
+
+    fn name_intersection(&self, i: IntersectionID) -> String {
+        // TODO When the name is missing, we could fallback on other OSM tags. See
+        // map_model::Road::get_name.
+        let road_names = self.network.intersections[&i]
```

Comment:
I thought about putting this in osm2streets directly, but because of the nuance around supporting multiple languages, decided not to yet.